security/pkg/nodeagent/caclient/providers/citadel/client.go

```diff
@@ -61,14 +62,20 @@ type TLSOptions struct {
 }
 
 // NewCitadelClient create a CA client for Citadel.
-func NewCitadelClient(opts *security.Options, tlsOpts *TLSOptions) (*CitadelClient, error) {
+func NewCitadelClient(opts *security.Options, tlsOpts *TLSOptions, watcher filewatcher.FileWatcher) (*CitadelClient, error) {
 	c := &CitadelClient{
 		tlsOpts:   tlsOpts,
 		opts:      opts,
 		provider:  caclient.NewCATokenProvider(opts),
 		usingMtls: atomic.NewBool(false),
 	}
 
+	if watcher != nil && c.tlsOpts.RootCert != "" {
+		if err := watcher.Add(c.tlsOpts.RootCert); err != nil {
```

Comment:
I think you need to watch xds_proxy used root cert as well, that one may not equal to this one